### DISCUSSION TITLE
00_pytorch_fundamentals_video.ipynb -

### DISCUSSION THREAD

**Discussion by 123321prog** [mrdbourke/pytorch-deep-learning#963] [Q&A]
![image](https://github.com/mrdbourke/pytorch-deep-learning/assets/74374482/d9a13311-6ba2-4c33-992c-376e837a5e79)

Is it supposed to be torch.size([1, 1, 4, 3])

**Comment by hemanthkotaprolu** [comment]
For me, I try to remember as:
- Vector as a list of scalars
- Matrix as a list of vectors
- Tensor as a list of matrix
- We can also think of high-dimensional tensors which can be thought of as a list of lower-dimensional tensors!!

So in your case, if we start with the dimension of vectors in the tensor. It is 3
1. Dimension of a vector is **3** => Number of scalars in each vector
2. Dimension of a matrix (list of vectors) which is basically how many vectors are there: **4** => Number of vectors in each matrix
3. Dimension of a tensor (list of matrix) which is **1** => Number of matrices in each tensor
4. Dimension of a higher level tensor (list of lower level tensors) is **1** => Number of lower dimensional tensors in each higher dimensional tensor

Finally keep everything (from right to left) => [1, 1, 4, 3]
